### PR TITLE
[SPIR-V] Sort basic blocks to match the SPIR-V spec

### PR DESCRIPTION
**Pull request by Keenuts**
The SPIR-V spec required basic blocks to respect some kind of ordering (A block dominating another cannot be after in the binary layout).

### CODE REVIEW

File: llvm/lib/Target/SPIRV/SPIRVPostLegalizer.cpp
Comment (michalpaszkowski) [inline llvm/lib/Target/SPIRV/SPIRVPostLegalizer.cpp:189]: One small suggestion: Adding `Order.reserve(MF.getNumBlocks());` would avoid rehashes. Though I suppose the difference would be practically non existent.

**Review by michalpaszkowski — APPROVED** [review]

File: llvm/lib/Target/SPIRV/SPIRVPostLegalizer.cpp
Comment (Keenuts) [reply llvm/lib/Target/SPIRV/SPIRVPostLegalizer.cpp:189]: Yes, that's a good idea! Updated the PR, thanks!

**Comment by Keenuts** [comment]
Waiting on the presubmit to finish, and will merge if fine.
(There is one failure upstream on the parser_overflow test, which is unrelated)

**Pull request merged**
